### ISSUE TITLE
setExtendedProp value param should be of type 'any'

### ISSUE THREAD

**Issue by maxwellt** [fullcalendar/fullcalendar#4679]
According to the documentation of the [Event Object](https://fullcalendar.io/docs/event-object), the property `extendedProps` is described as:

> A plain object holding miscellaneous other properties specified during parsing. Receives properties in the explicitly given extendedProps hash as well as other non-standard properties.

Yet on the `EventApi` object the method `setExtendedProps` has the following signature:

```
setExtendedProp(name: string, val: string): void;
```

I believe this should be:

```
setExtendedProp(name: string, value: any): void
```

Seeing as you can store miscellaneous properties there, it should not be limited to variables of type `string`.

If this is correct, please allow me to create a pull request, as I've never contributed to an open source project and this would be a very gentle introduction :-).

**Comment by acerix** [comment]
Do you get an error by calling that method if the the value is not a string? It's best if you can post a [runnable, stripped-down demonstration of the bug](http://fullcalendar.io/wiki/Reduced-Test-Cases/).

I believe you are correct that the type should be `any`.

**Comment by maxwellt** [comment]
@acerix , no error occurs but if you're using TSLint it will underline in red, which is how I saw that the expected type is 'string'.

**Comment by rozpuszczalny** [comment]
I'm confirming this. Setting, for example, `number` using `setExtendedProp` causes TS2345. Eg.:
```typescript
setNum(num: number, event: EventApi) {
  event.setExtendedProp('num', num); // TS2345
}
```

Ugly workaround:
```typescript
setNum(num: number, event: EventApi) {
  event.setExtendedProp('num', <string><unknown>num);
}
```

Looking at https://github.com/fullcalendar/fullcalendar/blob/f827f233581338ceb9723eabb7f895731498fd5b/src/core/structs/event-mutation.ts#L69 confirms that `setExtendedProp` `val` can be `any`.

**Comment by arshaw** [comment]
This was fixed in 4.3.0 though please start using 4.3.1
Releases: https://github.com/fullcalendar/fullcalendar/releases
